### PR TITLE
Global partners

### PR DESCRIPTION
**Pull request by amakarudze**
Changes in this PR
- Adds a `globalpartners` app for managing global partners and sending emails from the app
- Removes `Company` model from `jobboard/models.py`
- Alters field `company` of `Job` model to point to `GlobalPartner`

### CODE REVIEW

**Review by marksweb — APPROVED** [review]
This looks good.

My only feedback is just something I'd do differently - I'd store the sponsor level values as numbers rather than the strings. If any reporting were ever done on the data, numbers for how much people donate would be easier to query than the strings.

**Comment by amakarudze** [comment]
Closes #836 

> This looks good.
> 
> My only feedback is just something I'd do differently - I'd store the sponsor level values as numbers rather than the strings. If any reporting were ever done on the data, numbers for how much people donate would be easier to query than the strings.

Thanks @marksweb. I have changed the sponsor levels to integers.

File: globalpartners/models.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+from .emails import (
+    send_promotional_material_email,
+    send_prospective_sponsor_email,
+    send_renewal_email,
+    send_thank_you_email,
+)
+
+PATREON_LEVELS = [
+    (100, 100),
```
Comment (marksweb) [inline globalpartners/models.py:12]:
```suggestion
    (100, "$100/month"),
```

File: globalpartners/models.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    (250, 250),
```
Comment (marksweb) [inline globalpartners/models.py:13]:
```suggestion
    (250, "$250/month"),
```

File: globalpartners/models.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    (500, 500),
```
Comment (marksweb) [inline globalpartners/models.py:14]:
```suggestion
    (500, "$500/month"),
```

File: globalpartners/models.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    (1000, 1000),
```
Comment (marksweb) [inline globalpartners/models.py:15]:
```suggestion
    (1000, "$1,000/month"),
```

File: globalpartners/models.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+]
+
+ANNUAL_SPONSOR_LEVELS = [
+    (500, 500),
```
Comment (marksweb) [inline globalpartners/models.py:19]:
```suggestion
    (500, "Bronze ($500)"),
```

File: globalpartners/models.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    (1000, 1000),
```
Comment (marksweb) [inline globalpartners/models.py:20]:
```suggestion
    (1000, "Silver ($1000)"),
```

File: globalpartners/models.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    (2500, 2500),
```
Comment (marksweb) [inline globalpartners/models.py:21]:
```suggestion
    (2500, "Gold ($2,500)"),
```

File: globalpartners/models.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    (5000, 5000),
```
Comment (marksweb) [inline globalpartners/models.py:22]:
```suggestion
    (5000, "Platinum ($5,000)"),
```

File: globalpartners/models.py [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    (10000, 10000),
```
Comment (marksweb) [inline globalpartners/models.py:23]:
```suggestion
    (10000, "Diamond ($10,000)"),
```

**Comment by marksweb** [comment]
@amakarudze Sorry I should have given you an example rather than assume you'd know what I meant.

I'd make the choices;

```
[
    (100, "$100/month"),
    (250, "$250/month"),
    (500, "$500/month"),
    (1000, "$1,000/month"),
]
```

Then the database will store the integer, but where the user sees the choices they'll get the neat string to let them know the currency and the frequency of the payment.

I've added suggestions to the review to hopefully help out - you should just be able to accept each rather than actually making code changes.

I'd be good to also delete both migrations and generate the initial again so there's just 1.

**Review by marksweb — CHANGES_REQUESTED** [review]

**Comment by amakarudze** [comment]
Is there a reason for pre-commit fa

> @amakarudze Sorry I should have given you an example rather than assume you'd know what I meant.
> 
> I'd make the choices;
> 
> ```
> [
>     (100, "$100/month"),
>     (250, "$250/month"),
>     (500, "$500/month"),
>     (1000, "$1,000/month"),
> ]
> ```
> 
> Then the database will store the integer, but where the user sees the choices they'll get the neat string to let them know the currency and the frequency of the payment.
> 
> I've added suggestions to the review to hopefully help out - you should just be able to accept each rather than actually making code changes.
> 
> I'd be good to also delete both migrations and generate the initial again so there's just 1.

Yeah I thought about that and thought it would be me using it anyway and just used numbers all the way.

**Comment by marksweb** [comment]
@amakarudze The pre-commit failure appears to be a problem with isort

I think an update of the pre-commit hooks is in order. I'll raise a PR.

**Pull request merged**